windows/src/main/csharp/ch/cyberduck/ui/ViewModels/SynchronizedObservableObject.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright(c) 2002 - 2024 iterate GmbH. All rights reserved.
+// https://cyberduck.io/
+// 
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using System.ComponentModel;
+using System.Threading;
+
+namespace ch.cyberduck.ui.ViewModels;
+
+public abstract class SynchronizedObservableObject : ObservableObject
+{
+    private readonly SynchronizationContext sync = SynchronizationContext.Current;
```

Comment:
Capturing creating synchronization context to schedule PropertyChanged notifications back to UI thread